File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineExam20And21April2019/05.EasterEggs.js
function easterEggs(input) {
    let index = 0;
    let eggsCount = Number(input[index]);
    index++

    let redEggs = 0;
    let orangeEggs = 0;
    let blueEggs = 0;
    let greenEggs = 0;

    for(let i = 1; i <= eggsCount; i ++) {
        let eggColour = input[index];
        index++;

        switch(eggColour) {
            case 'red': redEggs++; break;
            case 'orange': orangeEggs++; break;
            case 'blue': blueEggs++; break;
            case 'green': greenEggs++; break;
        }
    }

    let maxEggs = Math.max(redEggs, orangeEggs, blueEggs, greenEggs);
    let colourOfMaxEggs = '';
    if(maxEggs === redEggs) {
        colourOfMaxEggs = 'red';
    } else if (maxEggs === orangeEggs){
        colourOfMaxEggs = 'orange';
    } else if (maxEggs === blueEggs) {
        colourOfMaxEggs = 'blue';
    } else {
        colourOfMaxEggs = 'green';
    }

    console.log(`Red eggs: ${redEggs}`);
    console.log(`Orange eggs: ${orangeEggs}`);
    console.log(`Blue eggs: ${blueEggs}`);
    console.log(`Green eggs: ${greenEggs}`);
    console.log(`Max eggs: ${maxEggs} -> ${colourOfMaxEggs}`);
}

easterEggs(["7",
"orange",
"blue",
"green",
"green",
"blue",
"red",
"green"])

easterEggs(["4",
"blue",
"red",
"blue",
"orange"]) 

File: Programming_Basics_with_Python/programming_basics_online_exam_20_and_21_april_2019/05._easter_eggs.py
painted_eggs_count = int(input())

red_eggs_count = 0
orange_eggs_count = 0
blue_eggs_count = 0
green_eggs_count = 0
for _ in range(painted_eggs_count):
    egg_colour = input()
    if egg_colour == 'red':
        red_eggs_count += 1
    elif egg_colour == 'orange':
        orange_eggs_count += 1
    elif egg_colour == 'green':
        green_eggs_count += 1
    elif egg_colour == 'blue':
        blue_eggs_count += 1

max_eggs_count = max(red_eggs_count, orange_eggs_count, blue_eggs_count, green_eggs_count)
max_eggs_colour = ''
if max_eggs_count == red_eggs_count:
    max_eggs_colour = 'red'
elif max_eggs_count == orange_eggs_count:
    max_eggs_colour = 'orange'
elif max_eggs_count == blue_eggs_count:
    max_eggs_colour = 'blue'
elif max_eggs_count == green_eggs_count:
    max_eggs_colour = 'green'

print(f'Red eggs: {red_eggs_count}')
print(f'Orange eggs: {orange_eggs_count}')
print(f'Blue eggs: {blue_eggs_count}')
print(f'Green eggs: {green_eggs_count}')
print(f'Max eggs: {max_eggs_count} -> {max_eggs_colour}')